### PR TITLE
[elasticexporter] Fix nil panic when setting custom headers

### PR DESCRIPTION
**Pull request by canhnt**
**Description:** <Describe what has changed.>
When setting custom headers in elasticexporter, such as:
```
  elasticsearch:
    endpoints: ['http://127.0.0.1:9211']
    logs_index: "example-index-01"
    headers:
      header-foo: bah
```

the latest version `0.63.0` throws the following panic:

```
panic: assignment to entry in nil map

goroutine 1 [running]:
net/textproto.MIMEHeader.Add(...)
	net/textproto/header.go:15
net/http.Header.Add(...)
	net/http/header.go:31
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter.newElasticsearchClient(0x40009c53b0, 0x4001240000)
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter@v0.63.0/elasticsearch_bulk.go:89 +0x138
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter.newLogsExporter(0x40009c53b0, 0x4001240000)
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter@v0.63.0/logs_exporter.go:49 +0x38
```

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** Added the unittest `TestExporter_New/create_with_custom_request_header` to cover this fix.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@canhnt Thanks for the contribution!

**Review by mx-psi — COMMENTED** [review]
Thanks for the PR! Could you [add a changelog note](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)?

**Comment by canhnt** [comment]
> Thanks for the PR! Could you [add a changelog note](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)?
@mx-psi : I've added the changelog file to the PR.

**Review by mx-psi — APPROVED** [review]

File: .chloggen/elasticexporter-fix-custom-header.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: elasticsearchexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fixed nil panic error when setting custom headers in elasticsearch exporter
+
+# One or more tracking issues related to the change
+issues: []
```
Comment (mx-psi) [inline .chloggen/elasticexporter-fix-custom-header.yaml:11]: You need to put an issue/PR here:
```suggestion
issues: [16017]
```

**Pull request merged**